src/bika/lims/content/contact.py

```diff
@@ -303,15 +303,15 @@ def _delUserFromGroup(self, username, group="Clients"):
         group = portal_groups.getGroupById(group)
         group.removeMember(username)
 
-    def _recursive_reindex_object_security(self, obj):
-        """Reindex object security after user linking
+    def _recursive_reindex_object(self, obj):
+        """Reindex object after user linking
         """
         if hasattr(aq_base(obj), "objectValues"):
             for child_obj in obj.objectValues():
-                self._recursive_reindex_object_security(child_obj)
+                self._recursive_reindex_object(child_obj)
 
-        logger.debug("Reindexing object security for {}".format(repr(obj)))
-        obj.reindexObjectSecurity()
+        logger.debug("Reindexing object {}".format(repr(obj)))
+        obj.reindexObject()
```

Comment:
`reindexObjectSecurity` forces the reindex of "security" indexes only,
- https://github.com/zopefoundation/Products.CMFCore/blob/361a30e0c72a15a21f88433b8d5fc49331f36728/src/Products/CMFCore/CMFCatalogAware.py#L128 and 
- https://github.com/zopefoundation/Products.CMFCore/blob/361a30e0c72a15a21f88433b8d5fc49331f36728/src/Products/CMFCore/CMFCatalogAware.py#L97

that are those involved on the internal filtering  searches by current user: 
- https://github.com/zopefoundation/Products.CMFCore/blob/361a30e0c72a15a21f88433b8d5fc49331f36728/src/Products/CMFCore/CatalogTool.py#L201-L208

The reindex of `allowedRolesAndUsers` index is way faster than the reindex of all object indexes. Is there any reason I am not aware of that requires the reindex of the whole object for the object to be displayed in listings?

Reply:
Ah, I think I mixed it and wondered why it did not work properly!
I will check this and change it to reindex only the `allowedRolesAndUsers` if it works out.